src/flows/pipes/QueryBuilder/CardList.tsx

```diff
@@ -125,23 +131,27 @@ const Card: FC<Props> = ({idx}) => {
     card.values.selected.length <= 1
 
   const valueSelect = val => {
-    let _vals = [...card.values.selected]
+    const _vals = [...card.values.selected]
     const index = _vals.indexOf(val)
 
-    if (index === -1) {
-      event('Query Builder Value Selected')
-      if (isCompliant) {
-        _vals = [val]
+    if (isCompliant) {
+      if (index === -1) {
+        event('Query Builder Value Selected')
+        selectMeasurement(val)
       } else {
-        _vals.push(val)
+        event('Query Builder Value Unselected')
+        selectMeasurement(null)
       }
+
+      return
```

Comment:
confused on this early return. Shouldn't we update the values for this idx anyways to clear it out? Or is that handled elsewhere?

Reply:
yeah, so all of the stuff done below is moved out into the context function. but the context  function is only used for  the  _measurement card

Reply:
Cool